packages/react-i18n/src/i18n.ts

```diff
@@ -377,6 +377,19 @@ export class I18n {
     return Boolean(easternNameOrderFormatter);
   }
 
+  hasTwelveHourClockFormat(locale?: string) {
+    const afternoonDateTime = new Date(2020, 1, 1, 13);
+    try {
```

Comment:
Under what situation will this throw? Is there a way to detect that with if/else conditionals rather than a try/catch?

Reply:
It will throw when an invalid locale is passed to `Intl.DateTimeFormat()`, like empty string or gibberish. 

On a related note, I noticed in the tests that there are test cases where locale is passed as 'unknown'. I tried using 'unknown' with the expectation that it would throw and default to 24 hour format but Intl parses 'unknown' as 'en-US'. 

I could possibly switch to if/else by validating the locale before I pass it to `Intl.DateTimeFormat`, but I figured a try/catch was the easier and more robust solution. Happy to change it if you have a preference and know a good way to validate a locale string.

Reply:
Thanks for the context.

So this function:
- returns true if the locale uses a 12h clock
- returns false if the locale uses a 24h clock or the locale is gibberish.

I think the throwing is valid so that we can determine the difference a garbage locale and one with a 24h clock. It looks like that is what occurs if you feed a garbage locale into `formatDate()`